### ISSUE TITLE
GPT-2 Neuron samples produce incorrect outputs (NaN) on Trn1, while official Neuron SDK BERT example works correctly

### ISSUE THREAD

**Issue by StevenLiuShiqi** [aws-neuron/aws-neuron-samples#119]
### Describe the bug

I observed incorrect and inconsistent behavior when running official GPT-2 and BERT notebooks from the aws-neuron-samples repository on a Trn1 instance using torch_neuronx.

Specifically:
1. The GPT-2 feature extraction sample produces all-NAN outputs on Neuron, while CPU outputs are non-zero.
2. GPT-2 language-model style inference (next-token logits) produces NaN logits on Neuron.
3. A BERT inference sample in the same aws-neuron-samples directory also fails to run correctly.
4. In contrast, a BERT example provided in the Neuron SDK repository runs correctly on the same environment, with CPU and Neuron outputs matching closely.

I suggest that the issue is not related to the Neuron environment or installation, but rather to GPT-2 support and/or sample compatibility with the current Neuron SDK.

### Model Name

gpt2, bert

### Describe the workload type

Inference only.

GPT-2:
1. Feature extraction (GPT2Model, hidden states)
2. Autoregressive next-token inference (GPT2LMHeadModel, logits)

BERT:
Sentence-pair classification

All models are compiled with torch_neuronx.trace using fixed input shapes.

### Instance Type

trn1.2xlarge

### Instance Type

i-02cd0974f129740cb

### Release version

aws-neuronx-collectives/unknown,now 2.29.41.0-681fef5f5 amd64 [installed]
aws-neuronx-dkms/unknown,now 2.25.4.0 all [installed]
aws-neuronx-oci-hook/unknown,now 2.13.52.0 amd64 [installed]
aws-neuronx-runtime-lib/unknown,now 2.29.40.0-f954cd7a5 amd64 [installed]
aws-neuronx-tools/unknown,now 2.27.33.0-5d9c0b901 amd64 [installed]
libneuronxla                  2.2.14584.0+06ac23d1
neuronx-cc                    2.22.12471.0+b4a00d10
neuronx-distributed           0.16.25997+f431c02e
neuronx-distributed-inference 0.7.14366+9adb71b8
torch                         2.9.0
torch-neuronx                 2.9.0.2.11.19912+e48cd891
torch-xla                     2.9.0
torchvision                   0.24.0
transformers                  4.36.0

### Reproduction Steps

1. GPT-2 feature extraction (official sample – incorrect output)

Notebook:
aws-neuron-samples/torch-neuronx/inference/hf_pretrained_gpt2_feature_extraction_on_trn1.ipynb

Behavior:
1. Uses GPT2Model for feature extraction
2. CPU output contains non-zero hidden states
3. Neuron output runs without crashing but returns all NANs

2. BERT example from Neuron SDK (works correctly)

Notebook:
aws-neuron-sdk/src/examples/pytorch/torch-neuronx/bert-base-cased-finetuned-mrpc-inference-on-trn1-tutorial.ipynb

Behavior:
1. CPU and Neuron logits match closely
2. No NaN or zero outputs
3. Predictions are identical

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Possible Solution

Based on these observations:
1. The Neuron environment and torch_neuronx installation appear to be correct.
2. Some notebooks in aws-neuron-samples may be out of sync with the current Neuron SDK.
3. GPT-2 models (both feature extraction and autoregressive inference) may have correctness or numerical stability issues under torch_neuronx.trace, leading to NaN or silent zero outputs.

Clarification on:
1. Current support status for GPT-2 models
2. Whether the GPT-2 samples are expected to work with the latest SDK
3. Recommended configuration or alternative approach for GPT-2 inference on Trn1

would be very helpful.

**Comment by musunita** [comment]
Hi StevenLiuShiqi,  GPT2-Sample NaN issue has been resolved by using the "--enable-saturate-infinity flag". For complete details, please refer to the [Neuron Compiler CLI Reference Guide](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/compiler/neuronx-cc/api-reference-guide/neuron-compiler-cli-reference-guide.html).
Regarding BERT Inference Sample: We've successfully tested the [BERT inference sample](https://github.com/aws-neuron/aws-neuron-samples/blob/master/torch-neuronx/inference/hf_pretrained_bert_inference_on_trn1.ipynb) on our latest Neuron SDK release, and inference completed successfully. To help us investigate the failure you're experiencing in Bert, could you please provide:
The specific error message or logs and reproduction steps
Your Neuron SDK version
Instance type and configuration details
Thank you for your patience as we work to resolve this issue.

**Comment by svskvar** [comment]
Closing this issue as it has been resolved. If you encounter this problem again, please feel free to reopen the issue or create a new one. Thank you.
